From da28ffa20d196f5c1e5fa660bc0bcd2301554f36 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Wed, 29 Jan 2025 18:45:59 +0100
Subject: [PATCH 1490/1695] dts: stm32: add DBGMCU secure debug mailbox on
 stm32mp21

Add DBGMCU secure debug mailbox on stm32mp21 platform. It is used by
Authenticated Debug Access Control (ADAC) protocol transport layer.

Only three registers of DBGMCU are used:
DBGMCU host authentication register (DBGMCU_DBG_AUTH_HOST at offset 0x100)
DBGMCU device authentication register (DBGMCU_DBG_AUTH_DEV at offset 0x104)
DBGMCU message read acknowledge authentication register
(DBGMCU_DBG_AUTH_ACK at offset 0x108).
That's why reg property address starts at 0x440a0100 (+0x100 compared to
DBGMCU base address), and reg property size covers the 3 32-bit registers.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 59eed808d4da619de2d77e7b99f325584d605a9b dts: stm32: add DBGMCU secure debug mailbox on stm32mp21]
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Change-Id: I87cd54f308e83e24899a70ea860d952806aad189
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/433589
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 core/arch/arm/dts/stm32mp211.dtsi | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index 75fa53fd6..0cb604972 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -693,6 +693,15 @@
 			clocks = <&rcc CK_BUS_SERC>;
 		};
 
+		dbgmcu_mbx: mailbox@440a0100 {
+			compatible = "st,stm32mp21-dbgmcu-mbx";
+			reg = <0x440a0100 0xc>;
+			interrupts = <GIC_SPI 208 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 209 IRQ_TYPE_EDGE_RISING>;
+			interrupt-names = "dbg_auth_wr", "dbg_auth_rd";
+			clocks = <&rcc CK_DBGMCU>;
+		};
+
 		rcc: clock-controller@44200000 {
 			compatible = "st,stm32mp21-rcc", "syscon";
 			reg = <0x44200000 0x10000>;
-- 
2.47.2

